### PR TITLE
Js unit testing

### PR DESCRIPTION
**Pull request by marcusramberg**
This pull request adds initial support for unit testing vue components.

npm test to run the test suite.

Note, requires npm install first to pull dependencies.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -17,3 +17,6 @@
 public/packed/
 assets/sass/.sass-cache/
 node_modules/
+.DS_Store
```
Comment (jhthorsen) [inline .gitignore]: :+1:

File: .travis.yml [hunk 2]
```diff
@@ -5,7 +5,11 @@ perl:
   - "5.16"
   - "5.10"
 env:
-  - "HARNESS_OPTIONS=j1 TEST_POD=1"
+  - "HARNESS_OPTIONS=j1 TEST_POD=1 TEST_PHANTOM=1"
+before_script:
+    - "export DISPLAY=:99.0" 
+    - "sh -e /etc/init.d/xvfb start"
+    - npm install
```
Comment (jhthorsen) [inline .travis.yml]: Awesome!

File: README.md [hunk 3]
```diff
@@ -1,29 +1,127 @@
...
+![Convos](http://convos.by/images/logo.png)
 
+Convos is the simplest way to use IRC. It is always online, and accessible to
+your web browser, both on desktop and mobile. Run in on your home server, or
+cloud service easily. It can be deployed to Docker-based cloud services, or
+you can just run it as a normal Mojolicious application, using any of the
+Deployment Guides.
 
- Convos is a multiuser chat application built with Mojolicious.
+![Screenshot](http://convos.by/images/screenshot.png)
```
Comment (jhthorsen) [inline README.md]: We need a new screenshot... Maybe we can use something from https://drive.google.com/open?id=0B_tCHIu9z7YSOTY5WnE5Mzk3M0E ?

File: README.md [hunk 4]
```diff
@@ -1,29 +1,127 @@
...
-```bash
-$ ./script/convos daemon;
-$ ./script/convos daemon --listen http://*:3000;
-```
+### Avatars
 
-And connect a browser to localhost:3000.
+The chat contains profile pictures which can be retrieved from Facebook or
+from gravatar.com.  The avatar used when talking with Convos is lovingly
+delivered by [Robohash.org](http://robohash.org/convos.by).
```
Comment (jhthorsen) [inline README.md]: "one-point-oh" branch does not support avatars. I'm not sure how to make that work nicely. It's a bit annoying when 99% of the users doesn't have a gravatar.

See also https://github.com/Nordaaker/convos/issues/252

File: README.md [hunk 5]
```diff
@@ -1,29 +1,127 @@
...
-To run Convos on a server, check out the relevant [Mojolicious Guides](http://mojolicious.org/perldoc/Mojolicious/Guides/Cookbook#DEPLOYMENT).
+### Include external resources
 
-[![Build Status](https://travis-ci.org/Nordaaker/convos.svg?branch=one-point-oh)](https://travis-ci.org/Nordaaker/convos)
+Links to images and video will be displayed inline. No need to click on the
+link to view the data.
+
+## Installation
+
+Convos requires a [Redis](http://redis.io/) server to function. If you are
```
Comment (jhthorsen) [inline README.md]: Nope. It doesn't.

I would very much appreciate if you could read and comment/change/copy/paste from https://github.com/Nordaaker/convos/blob/one-point-oh/lib/Convos/Guides/Running.pod

I think that goes for the rest of this README.

File: test/unit/user-register.spec.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+import Vue from 'vue'
+import UserRegister from '../../src/components/user-register.vue'
+
+describe('user-register.html.ep', () => {
+  it('should render correct contents', () => {
+    const vm = new Vue({
+      template: '<div><user-register></user-register></div>',
+      components: { UserRegister}
+    }).$mount()
+    expect(vm.$el.querySelector('.row p i').textContent).toBe('- Collaboration done right.')
```
Comment (jhthorsen) [inline test/unit/user-register.spec.js]: :+1:

**Comment by jhthorsen** [comment]
I would very much like to see these files go:

* src/App.vue
* src/components/Hello.vue
* src/main.js

What good does the .babelrc, .eslintrc files do?

I'm guessing package.json is to make "npm install" work?

Could you make ```t/js.t``` into automatically symlinking all the .html.ep files into ```src/components/``` and do ```END { remove_tree("src/components") }```?

**Comment by marcusramberg** [comment]
the babelrc is needed for the tests to pass. Removed the other files.

**Comment by marcusramberg** [comment]
All done.

File: t/js.t [hunk 7]
```diff
@@ -1,10 +1,17 @@
...
 
-my $res = `npm test`;
 
-diag $res;
+use File::Basename;
+@files = <templates/vue/*.html.ep>;
+foreach $file (@files) {
+  my $basefile = basename($file);
+  $basefile =~ s/\.html\.ep$/\.vue/;
+  symlink('../../' . $file, "src/components/" . $basefile);
```
Comment (jhthorsen) [inline t/js.t]: Why do you need "../../" ?

File: test/unit/user-register.spec.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
+/* global describe, it, expect */
+
+import Vue from "vue";
+import UserRegister from "../../src/components/user-register.vue";
+
+describe("user-register.html.ep", () => {
+    it("should render correct contents", () => {
```
Comment (jhthorsen) [inline test/unit/user-register.spec.js]: I don't get why this file has four spaces. All the other code that I run "jsfmt" on has two spaces, using https://github.com/Nordaaker/convos/blob/one-point-oh/.jsfmtrc

**Comment by jhthorsen** [comment]
Closing it regardless of the four spaces thing. Not important enough to leave this open :+1:

**Pull request merged**